.gitleaks.toml

```diff
@@ -0,0 +1,25 @@
+title = "Maistra Envoy gitleaks configuration"
+
+[[rules]]
+   description = "PRIVATE KEY CHECK PATHS"
+   regex = '''-----BEGIN ((EC|PGP|DSA|RSA|OPENSSH) )?PRIVATE KEY( BLOCK)?-----'''
+   tags = ["key"]
+   [rules.allowlist]
+      description = "test files"
+      paths = [
+         '''^test/common/grpc/service_key.json$''',
```

Comment:
Can we just whitelist everything under test or examples directories?

Reply:
+1, I agree, intuitively we don't have to worry about keys associated to tests and examples, so perhaps we could condense this list to something like `_test.cc` plus `^(test|examples)/.*`?

Reply:
@twghu kindly pinging this

Reply:
The problem relating to secret disclosure is in part the unintentional inclusion of files or content in files.  Listing each explicitly means we have identified the known problems.  Using a broad "anything in this folder" would allow for unintentional inclusions which by definition is the exact problem we are trying to avoid.

Reply:
Thanks for the explanation, with that this LGTM.
Considering that this wasn't obvious to me and John, perhaps that explanation is worth adding as a comment in the `.toml` file?